omod/src/main/java/org/openmrs/module/aijar/web/controller/AppointmentController.java

```diff
@@ -0,0 +1,47 @@
+package org.openmrs.module.aijar.web.controller;
+
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+@RequestMapping("/rest/ugandaemr/appointments")
+public class AppointmentController extends BaseRestController {
+
+    AdministrationService administrationService = Context.getAdministrationService();
+
+    @RequestMapping(method = RequestMethod.GET)
+    public int getNumberOfPatients(
+            @RequestParam(required = true, value = "nextAppointmentDate") Date nextAppointmentDate,
+            @RequestParam(value = "encounterType", defaultValue = "") String encounterType) {
+        List patients = null;
+
+        if (StringUtils.isEmpty(encounterType)) {
```

Comment:
@reagan-meant When the encounterType is specified you should be adding a where clause subselect 

AND obs.ecounter_id IN (SELECT e.encounter_id FROM encounter e INNER JOIN encounter_type ec ON e.encounter_type_id = ec.encounter_type_id AND e.uuid IN (encounterTypeUUid) 

You need to validate the column names and first run the queries on the database to ensure that they are well formed 

You can load the default OpenMRS demo data to test this out

Reply:
@ssmusoke I presume this is to go to line 38

Reply:
The line number does not matter the approach is all I care about